lib/crypto/c_src/ec.c

```diff
@@ -50,157 +50,183 @@ static EC_KEY* ec_key_new(ErlNifEnv* env, ERL_NIF_TERM curve_arg)
     BIGNUM *cofactor = NULL;
     EC_GROUP *group = NULL;
     EC_POINT *point = NULL;
+    int f_arity = -1;
+    const ERL_NIF_TERM *field;
+    int p_arity = -1;
+    const ERL_NIF_TERM *prime;
+    long field_bits;
 
     /* {Field, Prime, Point, Order, CoFactor} = Curve */
-    if (enif_get_tuple(env,curve_arg,&c_arity,&curve)
-	&& c_arity == 5
-	&& get_bn_from_bin(env, curve[3], &bn_order)
-	&& (curve[4] != atom_none && get_bn_from_bin(env, curve[4], &cofactor))) {
-
-	int f_arity = -1;
-	const ERL_NIF_TERM* field;
-	int p_arity = -1;
-	const ERL_NIF_TERM* prime;
-
-	long field_bits;
-
-	/* {A, B, Seed} = Prime */
-	if (!enif_get_tuple(env,curve[1],&p_arity,&prime)
-	    || !get_bn_from_bin(env, prime[0], &a)
-	    || !get_bn_from_bin(env, prime[1], &b))
-	    goto out_err;
-
-	if (!enif_get_tuple(env,curve[0],&f_arity,&field))
-	    goto out_err;
-
-	if (f_arity == 2 && field[0] == atom_prime_field) {
-	    /* {prime_field, Prime} */
-
-	    if (!get_bn_from_bin(env, field[1], &p))
-		goto out_err;
-
-	    if (BN_is_negative(p) || BN_is_zero(p))
-		goto out_err;
-
-	    field_bits = BN_num_bits(p);
-	    if (field_bits > OPENSSL_ECC_MAX_FIELD_BITS)
-		goto out_err;
-
-	    /* create the EC_GROUP structure */
-	    group = EC_GROUP_new_curve_GFp(p, a, b, NULL);
+    if (!enif_get_tuple(env, curve_arg, &c_arity, &curve))
+        goto err;
+    if (c_arity != 5)
+        goto err;
+    if (!get_bn_from_bin(env, curve[3], &bn_order))
+        goto err;
+    if (curve[4] != atom_none) {
+        if (!get_bn_from_bin(env, curve[4], &cofactor))
+            goto err;
+    }
 
-	} else if (f_arity == 3 && field[0] == atom_characteristic_two_field) {
+    /* {A, B, Seed} = Prime */
+    if (!enif_get_tuple(env, curve[1], &p_arity, &prime))
+        goto err;
+    if (!get_bn_from_bin(env, prime[0], &a))
+        goto err;
+    if (!get_bn_from_bin(env, prime[1], &b))
+        goto err;
+
+    if (!enif_get_tuple(env, curve[0], &f_arity, &field))
+        goto err;
+
+    if (f_arity == 2 && field[0] == atom_prime_field) {
+        /* {prime_field, Prime} */
+        if (!get_bn_from_bin(env, field[1], &p))
+            goto err;
+        if (BN_is_negative(p))
+            goto err;
+        if (BN_is_zero(p))
+            goto err;
+
+        field_bits = BN_num_bits(p);
+        if (field_bits > OPENSSL_ECC_MAX_FIELD_BITS)
+            goto err;
+
+        /* create the EC_GROUP structure */
+        if ((group = EC_GROUP_new_curve_GFp(p, a, b, NULL)) == NULL)
+            goto err;
+
+    } else if (f_arity == 3 && field[0] == atom_characteristic_two_field) {
 #if defined(OPENSSL_NO_EC2M)
-	    enif_raise_exception(env, atom_notsup);
-	    goto out_err;
+        enif_raise_exception(env, atom_notsup);
+        goto out_err;
```

Comment:
Got compilation error:
06134: ec.c: In function â€˜ec_key_newâ€™:
06135: ec.c:102:9: error: label â€˜out_errâ€™ used but not defined
06136:          goto out_err;
06137:          ^